src/main/resources/com/google/api/codegen/php/method_sample.snip

```diff
@@ -37,11 +36,37 @@
 @end
 
 @snippet incodeSamples(apiMethod)
-    @join sample : apiMethod.samples on BREAK.add(BREAK).add(BREAK).add("// Alternatively:").add(BREAK)
-        {@incodeSampleCall(apiMethod, sample)}
+    @join sample : apiMethod.samples on BREAK.add(BREAK).add(BREAK).add("// Alternatively:").add(BREAK).add(BREAK)
+        @let firstSample = apiMethod.samples.get(0)
+            # Print the set-up code for the first sample, and for
+            # subsequent samples where the code differs from the first
+            @if or(and(sample.callingForm == firstSample.callingForm, \
+                       sample.valueSet.id == firstSample.valueSet.id), \
+                sampleSetupDiffers(firstSample, sample))
+                {@incodeSampleSetup(apiMethod, sample)}
+            @end
+            {@incodeSampleCall(apiMethod, sample)}
+        @end
+    @end
+@end
+
+@private incodeSampleSetup(apiMethod, sample)
+    @if sample.initCode.lines
+        {@initCode(sample.initCode, apiMethod.apiVariableName)}
+
+    @end
+@end
+
+# This function returns true iff @incodeSampleSetup() would produce
+# different code for firstSample and secondSample.
+@private sampleSetupDiffers(firstSample, secondSample) horizontal
```

Comment:
Since this is used in Python too, maybe we could factor it out of both places and into `common/`?

Additionally, I would not   be opposed to also factoring the `if` condition in both languages so that the statement in both places reads `@if sampleNeedsSetup(sample, firstSample)`